### PR TITLE
Integer Factorization Sample Improvements

### PR DESCRIPTION
**Pull request by crazy4pi314**
This PR makes a number of small improvements to the readability of this sample, as well as removes all of the `fail` conditions.

### CODE REVIEW

File: samples/algorithms/integer-factorization/Shor.qs [hunk 1]
```diff
@@ -175,104 +168,213 @@ namespace Microsoft.Quantum.Samples.IntegerFactorization {
...
+    ///
+    /// # Input
+    /// ## modulus
+    /// The modulus which defines the residue ring Z mod `modulus`
+    /// in which the multiplicative order of `generator` is being estimated.
+    /// ## generator
+    /// The unsigned integer multiplicative order ( period )
+    /// of which is being estimated. Must be co-prime to `modulus`.
+    /// ## period
+    /// The estimated period from
```
Comment (cgranade) [inline samples/algorithms/integer-factorization/Shor.qs:338]: Looks like something went wrong here?

File: samples/algorithms/integer-factorization/Shor.qs [hunk 2]
```diff
@@ -175,104 +168,213 @@ namespace Microsoft.Quantum.Samples.IntegerFactorization {
...
+    /// in which the multiplicative order of `generator` is being estimated.
+    /// ## generator
+    /// The unsigned integer multiplicative order ( period )
+    /// of which is being estimated. Must be co-prime to `modulus`.
+    /// ## period
+    /// The estimated period from 
+    ///
+    /// # Output
+    /// A tuple of a flag bool and tuple that represent whether it 
+    /// found the factors and if it did a tuple of the factors.
```
Comment (cgranade) [inline samples/algorithms/integer-factorization/Shor.qs:342]: This is a bit confusing at the moment, perhaps rewording slightly could help
```suggestion
    /// A tuple of a flag indicating whether factors were found successfully,
    /// and a pair of integers representing the factors that were found.
    /// Note that the second output is only meaningful when the first
    /// output is `true`.
```

File: samples/algorithms/integer-factorization/Shor.qs [hunk 3]
```diff
@@ -175,104 +168,213 @@ namespace Microsoft.Quantum.Samples.IntegerFactorization {
...
+                return (false, (1,1));
+            }
+        } else {
+            // When period is odd we have to pick another generator to estimate
+            // period of and start over.
+            Message("Estimated period was odd, trying again.");
+            return (false, (1,1));
+        }
+    }
+}
```
Comment (cgranade) [inline samples/algorithms/integer-factorization/Shor.qs:380]:
```suggestion
}

```

File: samples/algorithms/integer-factorization/Untitled.ipynb [hunk 4]
```diff
@@ -0,0 +1,136 @@
+{
```
Comment (cgranade) [inline samples/algorithms/integer-factorization/Untitled.ipynb:1]: Should this be called "Untitled.ipynb"? I'd suggest a more specific name here.

Comment (crazy4pi314) [reply samples/algorithms/integer-factorization/Untitled.ipynb:1]: Good catch, I renamed it `Shor.ipynb` for consistency, and added to the README file list.

File: samples/algorithms/integer-factorization/Shor.qs (hunk 1 above)
Comment (crazy4pi314) [reply samples/algorithms/integer-factorization/Shor.qs]: Yeah I guess I spaced out mid sentence, fixed now!

**Comment by crazy4pi314** [comment]
I think this is good to go! Thanks everyone 💖

**Pull request merged**